generate_specs.py

```diff
@@ -162,30 +217,23 @@ def verify_one_function(func_name: str, llvm_analysis: LLVMAnalysis, out_file: P
         try:
             result = subprocess.run(command, shell=True, capture_output=True, text=True)
             if result.returncode == 0:
-                print(f"Verification for function {func_name} succeeded.")
-                success = True
+                print(
+                    f"Verification for function {func_name} succeeded (Number of attempts: {n + 1})"
+                )
                 processed_funcs.append(func_name)
-                break
-            filt_out = "\n".join([line for line in result.stdout.splitlines() if "FAILURE" in line])
-            repair_msg = (
-                f"Error while running verification for function {func_name}:\n"
-                f"STDOUT: {filt_out}\n"
-                f"STDERR: {result.stderr}\n"
-                "Please fix the error and re-generate the function with specifications."
-            )
+                return True
+            repair_msg = prompt_builder.build_repair_specification_prompt(parsec_function, result)
             print(repair_msg)
             conversation += [{"role": "user", "content": repair_msg}]
 
-            generate_spec(model, conversation, func_name, llvm_analysis, out_file)
+            write_new_spec_to_file(model, conversation, func_name, parsec_result, out_file)
 
         except Exception as e:
             print(f"Error running command for function {func_name}: {e}")
             break
 
-    if not success:
-        recover_from_failure()
-
-    return success
+    recover_from_failure()
+    return False
```

Comment:
Given that the function prints when it exits successfully, it should probably print when it exits unsuccessfully.